tools/ClassGen/NodeGen.cpp

```diff
@@ -484,6 +484,23 @@ int main(int argc, char **argv) {
                     "Weights[0] * Slice(0) + Weights[1] * Slice(1) + ... "
                     "It implies that len(Weights) == len(Indices).");
 
+  BB.newNode("SparseLengthsWeightedSumOffsets")
+      .addInput("Data")
+      .addInput("Weights")
+      .addInput("Indices")
+      .addInput("Offsets")
+      .addResultFromCtorArg()
+      .setDocstring(
+          "Gathers slices of the outer-most dimension of Data "
+          "indexed by Indices vector, and then accumulates them into "
+          "len(Offsets) entries: first slice between Offsets[0] and Offsets[1] "
+          "(or total length if there's only one elem in Offsets) are "
```

Comment:
@jfix71 and I were discussing this yesterday, when we compile for max length of all inputs including indices and weights then the end of the "real" part of this tensor if it's less than maximum will be larger than the legit inputs so the end indices won't be valid. I'm thinking it's ok because if we zero fill both tensors to the end then the sum of this section would be 0 due to the 0 weights. Depending on how it's implemented, this may be inefficient though. We could do something like add an extra offset at the end to mark the end of the last segment.

Reply:
Yeah actually nvm I think adding the extra offset in this glow op isn't feasible, I guess we have to rely on 0 weights

Reply:
Currently we have a branch in this op since there's no mark of the end..

Reply:
Not sure I understand what you mean